### DISCUSSION TITLE
createParamDecorator with services?

### DISCUSSION THREAD

**Discussion by kevin-mitchell** [MichalLytek/type-graphql#1251] [Q&A] [answered]
Looking at [this example](https://github.com/MichalLytek/type-graphql/blob/3733b1e57cb7ea8432420a17b3660c69187e0a09/examples/middlewares-custom-decorators/decorators/current-user.ts) I wonder how I might be able to use a service within this function.

For example, if my `CurrentUser` decorator needed to load some information from the database (via a repository or another service), based on some information in the `Context`, would there be a good way to do this?

Thank you!

**Comment by MichalLytek** [comment, marked as answer]
You should do all the heavy operation and populate the context before the GraphQL execution begins, just like you decode the jwt and put payload in context.

**Reply by kevin-mitchell** [reply]
This makes sense.

In my case I followed your advice, and loaded the user information to include in the request context closer to the bootstrapping of the app, before the graphql resolvers or other executions. In my case I directly used the DI `Container` to resolve the service I needed to load / validate users from request headers. I had to shift a few minor things around to do this in a nice way that avoided attempting to resolve a database repository before the database had bootstrapped, but this is specific to my case / implementation.

Thank you!
